### PR TITLE
Dev/took 214

### PR DESCRIPTION
**Pull request by Osman-Fatih-Cakir**

### CODE REVIEW

File: Editor/EditorLight.cpp [hunk 1]
```diff
@@ -299,29 +339,19 @@ namespace ToolKit
...
-      compMesh->m_material =
-        GetMaterialManager()->GetCopyOfUnlitColorMaterial();
-      compMesh->CalculateAABB();
-      AddComponent(mc);
+      // Mesh component for gizmo
+      m_gizmoMC = std::make_shared<MeshComponent>();
+      m_gizmoMC->ParamMesh().m_exposed = false;
 
-      // Gizmo
+      // Add gizmo meshes to mesh component
```
Comment (rbellek) [inline Editor/EditorLight.cpp:346]: This code block seems duplicated. If it is the same code line:153 please create a function and move it to the base class.
The function name is obvious: "Add gizmo meshes to mesh component".

File: Editor/EditorLight.cpp [hunk 2]
```diff
@@ -207,31 +238,21 @@ namespace ToolKit
...
-      compMesh->CalculateAABB();
-      AddComponent(mc);
+      // Mesh component for gizmo
+      m_gizmoMC = std::make_shared<MeshComponent>();
+      m_gizmoMC->ParamMesh().m_exposed = false;
 
       m_gizmo->InitGizmo(this);
 
-      // Gizmo
+      // Add gizmo meshes to mesh component
```
Comment (rbellek) [inline Editor/EditorLight.cpp:247]: This code block seems duplicated. If it is the same code line:153 please create a function and move it to the base class.
The function name is obvious: "Add gizmo meshes to mesh component".

File: Editor/EditorLight.cpp [hunk 3]
```diff
@@ -124,30 +145,21 @@ namespace ToolKit
...
-      mc->GetMeshVal()->CalculateAABB();
-      AddComponent(mc);
+      // Mesh component for gizmo
+      m_gizmoMC = std::make_shared<MeshComponent>();
+      m_gizmoMC->ParamMesh().m_exposed = false;
 
       m_gizmo->InitGizmo(this);
 
-      // Directional light gizmo
+      // Add gizmo meshes to mesh component
```
Comment (rbellek) [inline Editor/EditorLight.cpp:154]: This code block seems duplicated. If it is the same code line:247 please create a function and move it to the base class.
The function name is obvious: "Add gizmo meshes to mesh component".

**Review by rbellek — CHANGES_REQUESTED** [review]
Please fix duplicated code blocks.

**Review by afraidofdark — APPROVED** [review]

**Review by afraidofdark — APPROVED** [review]

**Pull request merged**